src/pages/groupComparison/pathwayMapper/GroupComparisonPathwayMapper.tsx

```diff
@@ -0,0 +1,392 @@
+    store: GroupComparisonStore;
+}
+
+/*function getMaxFrequencedGenes(
+    alterationRowData: AlterationEnrichmentRow[],
+    activeGroups: ComparisonGroup[] | undefined
+) {
+    var MaxFrequencedGenes: AlterationEnrichmentRow[] = [];
+    var usedGenes: number[] = [];
+    for (let j: number = 0; j < 10; j++) {
+        let idOfGene: number = -1;
+        let frequencyOfGene: number = 0;
+        for (let i: number = 0; i < alterationRowData.length; i++) {
+            let k: number = 0;
+            for (k = 0; k < j; k++) if (i === usedGenes[k]) break;
+            if (k < j) continue;
+            let maxFrequencyinGroups: number = 0;
+            let groupCount: number | undefined = activeGroups?.length;
+            if (groupCount !== undefined)
+                for (let k: number = 0; k < groupCount; k++)
+                    if (
+                        alterationRowData[i].groupsSet[k].alteredPercentage >
+                        maxFrequencyinGroups
+                    ) {
+                        maxFrequencyinGroups =
+                            alterationRowData[i].groupsSet[k].alteredPercentage;
+                    }
+            if (maxFrequencyinGroups > frequencyOfGene) {
+                frequencyOfGene = maxFrequencyinGroups;
+                idOfGene = i;
+            }
+        }
+        MaxFrequencedGenes.push(alterationRowData[idOfGene]);
+        usedGenes.push(idOfGene);
+    }
+}*/
+
+const DEFAULT_RULESET_PARAMS = getGeneticTrackRuleSetParams(true, true, true);
+
+@observer
+export default class GroupComparisonPathwayMapper extends React.Component<
+    IGroupComparisonPathwayMapperProps
+> {
+    clearMessage: (() => void) | undefined;
+    constructor(props: IGroupComparisonPathwayMapperProps) {
+        super(props);
+        makeObservable(this);
+
+        // @ts-ignore
+        import(/* webpackChunkName: "pathway-mapper" */ 'pathway-mapper').then(
+            (module: any) => {
+                this.PathwayMapperComponent = (module as any)
+                    .default as PathwayMapper;
+            }
+        );
+        if (this.props.store.isHighestFrequencedGenesCalculated === false) {
+            this.updateGenesOfInterestValue();
+            this.props.store.isHighestFrequencedGenesCalculated = true;
+        }
+    }
+    @observable.ref PathwayMapperComponent:
+        | PathwayMapper
+        | undefined = undefined;
+
+    @observable
+    isGeneSelectionPopupVisible: boolean = false;
+
+    @observable.ref
+    genes: AlterationEnrichmentRow[] = this.props.genesOfInterest;
+    @observable.ref
+    newlyAddedGenes: AlterationEnrichmentRow[] = this.props.genesOfInterest;
+    updateGenesOfInterestValue() {
+        let value: string = '';
+        let i: number = 0;
+        for (i = 0; i < this.props.store.maxFrequencedGenes.length; i++) {
+            value =
+                value +
+                this.props.store.maxFrequencedGenes[i].hugoGeneSymbol +
+                ' ';
+        }
```

Comment:
Avoid using for loops and string concat. Instead use `.map()` and `.join()` to construct your string.